src/libs/ascent/runtimes/expressions/ascent_blueprint_architect.cpp

```diff
@@ -992,6 +992,45 @@ field_histogram(const conduit::Node &dataset,
   return res;
 }
 
+//returns a node that is field1 + field2
+conduit::Node
+derived_field_add(const conduit::Node &dataset,
+                const std::string &field1,
+                const std::string &field2,
+		const std::string &out_field)
+{
+
+  conduit::Node res;
+  for(int i = 0; i < dataset.number_of_children(); ++i)
+  {
+    const std::string path1 = "fields/" + field1;
+    const std::string path2 = "fields/" + field2;
+    const conduit::Node &dom = dataset.child(i);
+    if(dom.has_path(path1) && dom.has_path(path2)) //has both
+    {
+      conduit::Node values;
+      values = derived_field_add_reduction(dom[path1], dom[path2]);
+      double *values = values["values"].value();
+      res[out_field].set(values); //need to preserve domain structure?
```

Comment:
@cyrush blueprint question for getting my data back in my result node. What rules do I need to be following here? Do I need to maintain the domain structure for my output field? If not, am I overwriting my out field with each set? Should I instead loop over all domains append the new field to a vector and then set it at the end?

Reply:
In this case, we can't pass const Node if we want to modify it (add a new field)

I think it would would be best is to return just the resulting Node field Node and insert it into the conduit tree at a higher level (in a custom filter)

We will also need to check association of the inputs (element or vertex) are the same and make sure they are both defined on the same topology. That info is in the field node along side the values. 

```
f1: 
  values : [..]
  topology : (string)
  association: (string)
```

The resulting field needs to propagate that info.

Reply:
Ah ok. This makes sense!